pkg/provider/azure_vmss.go

```diff
@@ -2359,3 +2353,89 @@ func (ss *ScaleSet) VMSSBatchSize(ctx context.Context, vmssName string) (int, er
 	klog.V(2).InfoS("Fetch VMSS batch size", "vmss", vmssName, "size", batchSize)
 	return batchSize, nil
 }
+
+func UpdateVMSSVMsInBatch(ctx context.Context, update map[string]armcompute.VirtualMachineScaleSetVM, batchSize int, patchVMFn func(ctx context.Context, instanceID string, vm *armcompute.VirtualMachineScaleSetVM) (*runtime.Poller[armcompute.VirtualMachineScaleSetVMsClientUpdateResponse], error)) <-chan error {
```

Comment:
It should not be an exported function. By the way, is it possible to put the logic inside arm client package just like before?

Reply:
UpdateVMSSVMsInBatch is only invoked here. it can't 'be used in vm agent pools. I think we can move function to azclient package if it is reused.
 I agree it should not be an exported one from business logic perspective. But it will be helpful if we want to write e2e test case. 
How do you think?